main.rb

```diff
@@ -1,4 +1,125 @@
 require_relative 'label'
 require_relative 'item'
+require_relative 'game'
+require_relative 'book'
+require_relative 'author'
+require 'date'
 
-def main_menu; end
+class App
+  attr_accessor :genre, :author, :source, :label, :publish_date, :cover_state, :choice, :labels, :books
+  attr_reader :id, :archived
+
+  def initialize
+    @labels = []
+    @books = []
+  end
+
+  def display_menu
+    puts 'What would you like to do?'
+    puts '1. List all books'
+    puts '2. List all genres'
+    puts '3. List all labels'
+    puts '4. List all authors'
+    puts '5. Add a book'
+    puts '6. Add a music album'
+    puts '7. Add a game'
+    puts '8. Quit'
+  end
+
+  def menu_choice
+    print 'Enter the number of your choice: '
+    gets.chomp.to_i
+  end
+
+  def handle_menu_choice(choice, app)
+    case choice
+    when 1
+      list_all_books
+    when 2
+      list_all_genres
+    when 3
+      list_all_labels
+    when 4
+      list_all_authors
+    when 5, 6, 7
+      create_item(choice, app)
+    when 8
+      return true
+    else
+      puts 'Invalid option entered'
+    end
+    false
+  end
+
+  def create_item(choice, app)
+    case choice
+    when 5
+      create_book(app)
+    when 6
+      create_album(app)
+    when 7
+      create_game(app)
+    end
+  end
+
+  def list_all_books
+    if books.empty?
+      puts 'No books found'
+    else
+      books.each_with_index do |book, index|
+        puts "#{index + 1}) Genre: #{book.genre} Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_all_labels
+    if labels.empty?
+      puts 'No labels found'
+    else
+      labels.each do |label|
+        puts "Title: #{label.title} | Color: #{label.color}"
+      end
+    end
+  end
+
+  def create_book(_app)
```

Comment:
I suggest this methods to be added in `app.rb`. Then we can create a new app in `main.rb`. This will help to reduce the number of lines of codes in `main.rb` which is responsible for entry point.

Reply:
@SimonGideon ty for your suggestion. Changes required implemented 👍